src/mono/wasm/runtime/assets.ts

```diff
@@ -76,9 +76,11 @@ export async function mono_download_assets(): Promise<void> {
                     asset.pendingDownloadInternal = asset.pendingDownload;
                     const waitForExternalData: () => Promise<AssetWithBuffer> = async () => {
                         const response = await asset.pendingDownloadInternal!.response;
-                        ++actual_downloaded_assets_count;
                         if (!headersOnly) {
                             asset.buffer = await response.arrayBuffer();
+                            ++actual_downloaded_assets_count;
+                        } else {
+                            ++actual_downloaded_assets_count;
                         }
```

Comment:
Why not `++actual_downloaded_assets_count;` after if-block if we are doint it anyway in both cases?